crates/verify/src/oklink/flatten.rs

```diff
@@ -0,0 +1,110 @@
+use super::{OklinkSourceProvider, VerifyArgs};
+use eyre::{Context, Result};
+use foundry_block_explorers::verify::CodeFormat;
+use foundry_compilers::{
+    artifacts::{BytecodeHash, Source},
+    AggregatedCompilerOutput, CompilerInput, Project, Solc,
+};
+use semver::{BuildMetadata, Version};
+use std::{collections::BTreeMap, path::Path};
+
+#[derive(Debug)]
+pub struct OklinkFlattenedSource;
+impl OklinkSourceProvider for OklinkFlattenedSource {
```

Comment:
Can we instead push all of this to the oklink verifier impl instead?

I don't think we need the traits